plutus-contract/src/Language/Plutus/Contract/Trace.hs

```diff
@@ -109,8 +111,9 @@ type ContractTrace s e m a = StateT (ContractTraceState s (TraceError e) a) m
 
 data WalletState s =
     WalletState
-        { walletEvents   :: Seq (Event s)
-        , walletHandlers :: Handlers s
+        { walletEvents         :: Seq (Event s)
+        , walletHandlers       :: Handlers s
+        , walletSigningProcess :: SigningProcess
```

Comment:
This feels to me like it's in the wrong place, shouldn't this be in `plutus-emulator` somwhere?

Reply:
Yeah true I'll move it to `Wallet.Emuator.Types.WalletState`

Reply:
The idea of a signing process that's separate from the wallet doesn't exist in the emulator though. Everything is part of `WalletAPI` so I don't think we should make `WalletAPI` even more powerful by adding stuff to it. `Wallet.API.signTxn` even has a note saying "In the future this won't be part of WalletAPI to allow the signing to be handled by a different process".